scene/resources/environment.cpp

```diff
@@ -891,23 +894,41 @@ float Environment::get_adjustment_saturation() const {
 	return adjustment_saturation;
 }
 
-void Environment::set_adjustment_color_correction(const Ref<Texture2D> &p_ramp) {
-	adjustment_color_correction = p_ramp;
+void Environment::set_adjustment_color_correction(Ref<Texture> p_color_correction) {
+	adjustment_color_correction = p_color_correction;
+	Ref<GradientTexture> grad_tex = p_color_correction;
+	if (grad_tex.is_valid()) {
```

Comment:
This stuff isn't necessary. Since we just pass the RID to the rendering device, all updates happen automatically.

Reply:
But gradient changing doesn't work. I guess we can document that you need to restart the editor to modify the graident..

Reply:
Closing, please reopen if you have arguments why we need to restart the editor to change colours on the gradient colours list.

Reply:
Did you test it out? You can update the gradient and everything updates automatically. There is no point in sending the same RID to the server over and over again. 

Take a look at the 3.2 code, it worked the same way.

Reply:
Yes I did. I can give you a video.

Reply:
https://imgur.com/a/gsVy0zq Here are the two variations of the changes. One has this code removed.

Reply:
The only one that works is when I connect the gradient changed signal to changing the adjustments.